components/prism-psl.js

```diff
@@ -0,0 +1,40 @@
+Prism.languages.psl = {
+	'comment': {
+		pattern: /#.*/,
+		greedy: true
+	},
+	'string': {
+		pattern: /"(?:\\.|[^\\"])*"/,
+		greedy: true,
+		inside: {
+			'symbol': /\\[ntrbA-Z"\\]/
+		}
+	},
+	'heredoc-string': {
+		pattern: /<<<([a-zA-Z_]\w*)[\r\n](?:.*[\r\n])*?\1\b/,
+		alias: 'string',
+		greedy: true
+	},
+	'keyword': /\b(?:__multi|__single|case|default|do|else|elsif|exit|export|for|foreach|function|if|last|line|local|next|requires|return|switch|until|while|word)\b/,
+	'constant': /\b(?:ALARM|CHART_ADD_GRAPH|CHART_DELETE_GRAPH|CHART_DESTROY|CHART_LOAD|CHART_PRINT|EOF|FALSE|False|false|NO|No|no|OFFLINE|OK|PSL_PROF_LOG|R_CHECK_HORIZ|R_CHECK_VERT|R_CLICKER|R_COLUMN|R_FRAME|R_ICON|R_LABEL|R_LABEL_CENTER|R_LIST_MULTIPLE|R_LIST_MULTIPLE_ND|R_LIST_SINGLE|R_LIST_SINGLE_ND|R_MENU|R_POPUP|R_POPUP_SCROLLED|R_RADIO_HORIZ|R_RADIO_VERT|R_ROW|R_SCALE_HORIZ|R_SCALE_VERT|R_SPINNER|R_TEXT_FIELD|R_TEXT_FIELD_LABEL|R_TOGGLE|TRIM_LEADING|TRIM_LEADING_AND_TRAILING|TRIM_REDUNDANT|TRIM_TRAILING|TRUE|True|true|VOID|WARN)\b/,
+	'variable': /\b(?:errno|exit_status|PslDebug)\b/,
+	'builtin-function': {
```

Comment:
I not sure about the aliases you gave to built-in functions and user-defined functions. They are technically correct but they aren't consistent with how Prism uses them. In other languages, we use `builtin` to refer to either built-in types, functions, or (in rare cases) variables. Similarly, `variables` is usually used for _special_ variables like you did for `errno` and not "normal" variables and especially not function variables. They are always exceptions to these patterns but I try to keep the usage of token names consistent across languages.

For this language, I suggest the following: Keep the token names but change the aliases. The names are very descriptive, so I think they should be kept. 
`user-function` should definitely get a `function` alias and lose the `variable` alias. This will make it consistent with (most of) Prism's other languages.
I'm not sure about `builtin-function`. The `function` alias is fitting but so is `builtin`. I'm unsure as to whether `builtin` is best to be added as an alias or whether it should replace `function`.

Why did you choose these aliases in the first place?

Reply:
Good question!
At first, I went with the same as you suggested, i.e. `'builtin'` token for built-in functions, and `'function'` for user-defined functions.
The problem is that *builtin* and *string* tokens are colored the same way in most themes. So the semantic of a simple statement like below is not conveyed clearly (everything:
```perl
# PERL
print("Hello, World!");
```
Prism Default Theme, Dark and Funky:
![image](https://user-images.githubusercontent.com/32521698/106007798-f4a99180-60b6-11eb-90af-a4a33575c26c.png)
![image](https://user-images.githubusercontent.com/32521698/106007900-1145c980-60b7-11eb-9033-db10080e1b2f.png)
![image](https://user-images.githubusercontent.com/32521698/106007937-1f93e580-60b7-11eb-9f4f-f1855e85570c.png)

Twilight and Tomorrow Night themes do use different colors, though:
![image](https://user-images.githubusercontent.com/32521698/106008182-55d16500-60b7-11eb-962c-e4154894e98b.png)
![image](https://user-images.githubusercontent.com/32521698/106008225-5ec23680-60b7-11eb-8fa2-6dbd2004db7d.png)

(Note: in **prism-perl.js**, `print` is a *keyword*, not a *builtin*)

Anyway, so that's the reason why I bent the rules a little to have a better rendering. But I agree with your view on having a consistent display of various token types across different languages.

I you prefer me go revert back to "builtin" and "function", I will follow your suggestion (and make some CSS tweaks in my project). Just let me know!

Reply:
Hmmm, `user-function`s should definitely have a `function` alias, I won't budge on that, but I see the problem with `builtin-function`s.

> (Note: in prism-perl.js, print is a keyword, not a builtin)

Does the `keyword` alias for `builtin-function` work for you?

Otherwise, I'd say that we just use the `builtin` and `function` alias for `builtin-function` and `user-function` respectively. The token names should make it rather simple for users to adjust themes.

Reply:
I prefer to stick to `builtin` and `function` and not use `keyword` to avoid mixing things that are semantically different. If the user (like me) is unhappy with the color of a theme, it's pretty simple to customize the color for `builtin`.
Alright, it's settled, then! Pushing a new commit now.